cpp/daal/src/algorithms/dbscan/dbscan_distributed_input_types.cpp

```diff
@@ -1197,14 +1198,17 @@ services::Status DistributedInput<step12Local>::check(const daal::algorithms::Pa
     const size_t nQueriesBlocks = dcOrders->size();
     DAAL_CHECK_EX(nQueriesBlocks > 0, ErrorIncorrectNumberOfInputNumericTables, ArgumentName, step12PartialOrdersStr());
 
-    for (size_t i = 0; i < nQueriesBlocks; i++)
+    if (NumericTable::cast((*dcOrders)[0])->getNumberOfRows() != 0)
```

Comment:
it is reasonable to put this check inside loop and just skip tables with 0 rows.